geowebcache/rest/src/main/java/org/geowebcache/rest/layers/TileLayerRestlet.java

```diff
@@ -335,6 +345,16 @@ private void doDelete(Request req, Response resp) throws RestletException, GeoWe
         String layerName = ServletUtils.URLDecode((String) req.getAttributes().get("layer"),
                 "UTF-8");
         findTileLayer(layerName, layerDispatcher);
+        // delete cached tiles first in case a blob store
+        // uses the configuration to perform the deletion
+        StorageException storageBrokerDeleteException = null;
+        try {
+            storageBroker.delete(layerName);
```

Comment:
It bothers me a little that this is happening in the REST implementation.  It seems like it should be an inherent behaviour of the objects that when the layer is deleted from a configuration, the storage for it is deleted.  If we added another interface for deleting layers such as an improved web UI, it would need to match this behaviour which the implementer might not consider.

There's probably a similar problem in that the layer, configuration, and storage implementations all have multiple implementations. We probably need a more comprehensive event system to handle the situation cleanly, possibly with pre-delete and post-delete events.  Admittedly that would be a lot of work and outside the scope of a bug fix.

If this was just the least bad place to put it given the lack of comprehensive events, I think it's worth a comment to explain why it's here.